packages/dev/core/src/Gizmos/planeRotationGizmo.ts

```diff
@@ -300,14 +310,15 @@ export class PlaneRotationGizmo extends Gizmo implements IPlaneRotationGizmo {
                 if (this.updateGizmoRotationToMatchAttachedMesh) {
                     // Rotate selected mesh quaternion over fixed axis
                     nodeQuaternion.multiplyToRef(amountToRotate, nodeQuaternion);
+                    // recompose matrix
+                    this.attachedNode.getWorldMatrix().copyFrom(Matrix.Compose(nodeScale, nodeQuaternion, nodeTranslation));

```

Comment:
I think you can use `Matrix.ComposeToRef` to avoid GC.

However, is that working as expected? The world matrix of `attachedNode` will be recreated as soon as its position/rotation/scale changes and will overwrite the value copied from the line above? Or is it that those properties of `attachedNode` are never updated directly?

Reply:
TRS is updated in `_matrixChanged` depending of the nature of the node.